components/playlist/browser/media_detector_component_manager.cc

```diff
@@ -79,6 +91,102 @@ MediaDetectorComponentManager::MediaDetectorComponentManager(
+       base::BindRepeating([](const GURL& url) {
+         // "https://bbcgoodfood.com/videos/foo"
+         if (url.path_piece() == "/") {
+           return false;
+         }
+
+         auto paths = SplitPaths(url);
+         return paths.size() == 2 && paths.front() == "videos";
+       })},
+      {net::SchemefulSite(GURL("https://bitchute.com")),
+       base::BindRepeating([](const GURL& url) {
+         // https://www.bitchute.com/video/XXXXXXXXXXXX/
+         if (url.path_piece() == "/") {
+           return false;
+         }
+
+         auto paths = SplitPaths(url);
+         return paths.size() == 2 && paths.front() == "video";
+       })},
+      {net::SchemefulSite(GURL("https://brighteon.com")),
+       base::BindRepeating([](const GURL& url) {
+         // GUID format
+         // https://www.brighteon.com/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+         if (url.path_piece() == "/") {
+           return false;
+         }
+
+         auto paths = SplitPaths(url);
+         if (paths.size() != 1) {
+           return false;
+         }
+
+         const auto guid_parts = SplitString(paths.front(), "-");
+         if (guid_parts.size() != 5) {
+           return false;
+         }
+
+         return guid_parts.at(0).length() == 8 &&
+                guid_parts.at(1).length() == 4 &&
+                guid_parts.at(2).length() == 4 &&
+                guid_parts.at(3).length() == 4 &&
+                guid_parts.at(4).length() == 12;
+       })},
+      {net::SchemefulSite(GURL("https://marthastewart.com")),
+       base::BindRepeating([](const GURL& url) {
+         // TODO(sko) Can't tell a page would have media from url.
+         return false;
+       })},
+      {net::SchemefulSite(GURL("https://rumble.com/")),
+       base::BindRepeating([](const GURL& url) {
+         // Starts with vXXXXXX (6 chars) and followed by title of video
+         // https://rumble.com/vXXXXXX-foo-bar-baz.html
+         if (url.path_piece() == "/") {
+           return false;
+         }
+
+         auto paths = SplitPaths(url);
+         if (paths.size() != 1) {
+           return false;
+         }
+
+         const auto media_page_parts = SplitString(paths.front(), "-");
+         return media_page_parts.size() > 1 &&
+                media_page_parts.front().size() == 7 &&
+                media_page_parts.front().at(0) == 'v';
+       })},
+      {net::SchemefulSite(GURL("https://ted.com")),
+       base::BindRepeating([](const GURL& url) {
+         // https://www.ted.com/talks/frances_s_chance_are_insect_brains_the_secret_to_great_ai
+         if (url.path_piece() == "/") {
+           return false;
+         }
+
+         auto paths = SplitPaths(url);
+         return paths.size() == 2 && paths.front() == "talks";
+       })},
+      {net::SchemefulSite(GURL("https://vimeo.com")),
+       base::BindRepeating([](const GURL& url) {
+         // https://vimeo.com/1234567  (5-9 digits)
+         return false;
+       })},
+  };
```

Comment:
Hmm, that's quite unscalable, isn't it? I mean, doing this one by one for all the sites that might (or might not) have `MediaSource`-based `<video>`s on them seems brittle. We're always going to forget about something, e.g. even here the checker for YouTube doesn't cover YouTube Shorts (`https://www.youtube.com/shorts/...`), or the structure of a URL changes, or a site stops using the `MediaSource` API, etc.

I think it should be the [playlist-component](https://github.com/brave/playlist-component)'s responsibility to tell whether a page has to be loaded in the background, or not, that is, if it has a `blob:` backed by a `MediaSource` as the `src` of its `<video>`.

How about extending `index.js`'s functionality (which already runs in a private execution context). Something along these lines:
```js
function waitUntil(predicate) {
  const executor = (resolve) =>
    predicate() ? resolve() : setTimeout(() => executor(resolve), 50)
  return new Promise(executor)
}

async function isMediaSourceObjectURL(src) {
  if (!src.startsWith('blob:')) {
    return false
  }

  const controller = new AbortController()
  const signal = controller.signal
  let timeout
  const maybeAbortFetch = new Promise(resolve =>
    timeout = setTimeout(() => {
      console.log('aborting fetch')
      resolve(false)
      controller.abort()
    }, 500)
  )

  return Promise.race([
    new Promise(resolve => {
      fetch(src, {
          signal
        })
        .then(response => {
          console.log(response)
          resolve(false)
        })
        .catch(error => {
          console.log(error)
          resolve(true)
        })
        .finally(() => {
          clearTimeout(timeout)
        })
    }),
    maybeAbortFetch
  ])
}

window.addEventListener('DOMContentLoaded', () => {
  const video = document.querySelector('video')
  if (video) {
    waitUntil(() => video.src).then(() => isMediaSourceObjectURL(video.src)
      .then(flag => console.log(
        `(${window.location.href}) ${video.src} is${flag ? ' ' : ' not '}a MediaSource blob URL.`
      )))
  }
})
```
And then passing back the result as:
```js
{
  name,
  src,
  'srcIsMediaSourceObjectURL': isMediaSourceObjectURL(src),
  ...
}
```
Above assumes that `fetch()` doesn't throw a `TypeError` with the message `Failed to fetch` in cases other than when `blob:` refers to a `MediaSource`. Even if that's not guaranteed (which we should be able to check [here](https://fetch.spec.whatwg.org)), it should still be a very good approximation.

Alternatively, a more precise solution would be to monkey patch `URL.createObjectURL()` (like [here](https://github.com/brave-experiments/youtube-downloader/blob/b20f7dc9231b0f0f5c160d2a309da9c6c7c97a6e/js/content_main.js#L294-L302)), cache the result of the call if the object passed in is a `MediaSource`, and do a string comparison with the `src` of the `<video>`. This, however, we'd need to inject into `MAIN`.

Reply:
I can't agree more with you. Let me try that.

Reply:
It seems that `async/await` is not supported when using `ExecuteJavascript` from browser side. I'm trying to move detection(script execution) to renderer side. this is what I planned, disregarding this problem.

Reply:
Update: async/await seems to be not supported from the renderer side too.

```
[27116:259:1219/090728.076120:ERROR:playlist_js_handler.cc(150)] "await is not defined"
```

Reply:
> It seems that async/await is not supported when using ExecuteJavascript from browser side

This is not correct. See components/script_injector

Reply:
> > It seems that async/await is not supported when using ExecuteJavascript from browser side
> 
> This is not correct. See components/script_injector

Yes, I header about it from the security review.